sdks/builds/llvm.mk

```diff
@@ -23,26 +32,35 @@ _llvm_$(1)_CONFIGURE_ENVIRONMENT= \
 	CXXFLAGS="$$(_llvm_$(1)_CXXFLAGS)" \
 	LDFLAGS="$$(_llvm_$(1)_LDFLAGS)"
 
-_llvm_$(1)_CONFIGURE_FLAGS= \
-	--build=$$(if $$(filter $$(UNAME),Darwin),$(2)-apple-darwin10,$$(if $$(filter $$(UNAME),Linux),x86_64-linux-gnu,$$(error "Unknown UNAME='$$(UNAME)'"))) \
-	--cache-file=$$(TOP)/sdks/builds/llvm-$(1).config.cache \
-	--prefix=$$(TOP)/sdks/out/llvm-$(1) \
-	--enable-assertions=no \
-	--enable-optimized \
-	--enable-targets="arm,aarch64,x86" \
-	$$(if $$(filter $$(UNAME),Darwin),--enable-libcpp)
+_llvm_$(1)_CMAKE_FLAGS = \
+	$(llvm_base_CMAKE_FLAGS) \
+	-DCMAKE_INSTALL_PREFIX=$$(TOP)/sdks/out/llvm-$(1) \
+	$$(llvm_$(1)_CMAKE_FLAGS) \
+	$(LLVM_SRC)
 
 .stamp-llvm-$(1)-toolchain: | $$(LLVM_SRC)
 	touch $$@
 
-.stamp-llvm-$(1)-configure: $$(LLVM_SRC)/configure
+.stamp-llvm-$(1)-configure:
 	mkdir -p $$(TOP)/sdks/builds/llvm-$(1)
-	cd $$(TOP)/sdks/builds/llvm-$(1) && $$< $$(_llvm_$(1)_CONFIGURE_ENVIRONMENT) $$(_llvm_$(1)_CONFIGURE_FLAGS)
+	cd $$(TOP)/sdks/builds/llvm-$(1) && cmake $$(_llvm_$(1)_CMAKE_FLAGS)
 	touch $$@
 
 .PHONY: package-llvm-$(1)
+
+ifneq ($(NINJA),)
+build-custom-llvm-$(1):
+	ninja -C $$(TOP)/sdks/builds/llvm-$(1)
```

Comment:
This could be `$(if $(NINJA),ninja,make)`, allowing to remove the duplicate https://github.com/mono/mono/pull/8260/files#diff-e19948cc4c89121d1aa1068fd3d60b01R52 and https://github.com/mono/mono/pull/8260/files#diff-e19948cc4c89121d1aa1068fd3d60b01R58. Same at https://github.com/mono/mono/pull/8260/files#diff-e19948cc4c89121d1aa1068fd3d60b01R56

Reply:
It could, but these are different tools which happen to have a common subset of command line options.

Reply:
We could probably use cmake --build to invoke the appropriate build tool.

Reply:
It's mostly to avoid repetition, so I would prefer the `$(if $(NINJA),ninja,make)` but that's not-blocking to the PR.